Lib/test/test_httpservers.py

```diff
@@ -553,6 +558,202 @@ def test_html_escape_filename(self):
+            self.assertEqual(gzip.decompress(response.read()), self.data)
+
+        # Big files
+        for ext in self.compressible_ext:
+            response = self.request(self.base_url + '/test_big.{}'.format(ext),
+                headers={'Accept-Encoding': 'gzip'})
+            self.assertTrue('Content-Encoding' in response.headers)
+            # on HTTP/1.0 Chunked Tranfer Encoding is not supported
+            self.assertFalse('Transfer-Encoding' in response.headers)
+            self.assertEqual(gzip.decompress(response.read()),
+                self.repeat * self.data)
+
+    def test_header_set_to_wrong_value_supported_extension(self):
+        # Content-Encoding header set to a value that doesn't include gzip,
+        # compressible file extension
+        for ext in self.compressible_ext:
+            # quality value set to 0 means that the value is not supported
+            response = self.request(self.base_url + '/test.{}'.format(ext),
+                headers={'Accept-Encoding': 'gzip;q=0'})
+            self.assertFalse('Content-Encoding' in response.headers)
+
+            response = self.request(self.base_url + '/test.{}'.format(ext),
+                headers={'Accept-Encoding': '*;q=0'})
+            self.assertFalse('Content-Encoding' in response.headers)
+
+            response = self.request(self.base_url + '/test.{}'.format(ext),
+                headers={'Accept-Encoding': 'dummy'})
+            self.assertFalse('Content-Encoding' in response.headers)
+
+    def test_user_defined_compressions(self):
+        # test with encoding "bzip2" instead of "gzip"
+
+        # Write a generator for (non-standard) bzip2 compression encoding.
+        def _bzip2_producer(fileobj):
+            bufsize = 2 << 17
+            producer = bz2.BZ2Compressor()
+            with fileobj:
+                while True:
+                    buf = fileobj.read(bufsize)
+                    if not buf: # end of file
+                        data = producer.flush()
+                        if data:
+                            yield data
+                        yield b''
+                        return
+                    data = producer.compress(buf)
+                    if data:
+                        yield data
+
+        # update dictionary "compressions" to support bzip2
+        self.request_handler.compressions.update(bzip2=_bzip2_producer)
+
+        for ext in self.compressible_ext:
+            response = self.request(self.base_url + '/test.{}'.format(ext),
+                headers={'Accept-Encoding': 'bzip2'})
+            self.assertTrue('Content-Encoding' in response.headers)
+            self.assertEqual(bz2.decompress(response.read()), self.data)
+
+class HTTPCompressionChunkedTransferTestCase(HTTPCompressionTestCase):
+
+    class request_handler(NoLogRequestHandler, SimpleHTTPRequestHandler):
+
+        protocol_version = 'HTTP/1.1'
+        default_request_version = 'HTTP/1.1'
```

Comment:
Doubt _default_request_version_ is necessary

Reply:
Done in 50f0a85